ipv8/bootstrapping/dispersy/bootstrapper.py

```diff
@@ -33,41 +30,41 @@ def __init__(self,
         """
         Create a new bootstrapper without resolving DNS addresses or initializing the blacklist.
         """
+        super().__init__()
+
         self.ip_addresses = [UDPv4Address(*addr) for addr in ip_addresses]
         self.dns_addresses = dns_addresses
 
         self.bootstrap_timeout = bootstrap_timeout
 
         self.last_bootstrap: float = 0
 
-    def resolve_dns_bootstrap_addresses(self) -> None:
+    async def resolve_dns_bootstrap_addresses(self) -> None:
         """
         Resolve the bootstrap server DNS names defined in ``dns_addresses`` and insert them into ``ip_addresses``.
         """
-        def resolve_addresses(dns_names: Iterable[Address], ip_addresses: list[Address]) -> None:
-            current_addresses = ip_addresses[:]  # Copy the existing addresses (don't loop through our additions)
-            for (address, port) in dns_names:
-                try:
-                    resolved_address = UDPv4Address(gethostbyname(address), port)
-                    if resolved_address not in current_addresses:
-                        # NOTE: append() is thread-safe. Don't call remove() here!
-                        ip_addresses.append(resolved_address)
-                except OSError:
-                    logging.info("Unable to resolve bootstrap DNS address (%s, %d)", address, port)
-
-        resolution_thread = Thread(name="resolve_dns_bootstrap_addresses",
-                                   target=resolve_addresses,
-                                   args=(self.dns_addresses, self.ip_addresses),
-                                   daemon=True)
-        resolution_thread.start()
-
-    def initialize(self, overlay: Community) -> Future:
+        loop = asyncio.get_running_loop()
+
+        async def resolve_address(host: str, port: int) -> None:
+            try:
+                ip = await loop.run_in_executor(None, gethostbyname, host)
+                address = UDPv4Address(ip, port)
+                if address not in self.ip_addresses:
+                    self.ip_addresses.append(address)
+            except OSError:
+                logging.info("Unable to resolve bootstrap DNS address (%s, %d)", host, port)
+
+        await asyncio.gather(*[resolve_address(*address) for address in self.dns_addresses])
+
+    async def initialize(self, overlay: Community) -> bool:
         """
         Initialize this bootstrapper for the given Community, settings its blacklist.
         """
-        overlay.network.blacklist.extend(self.ip_addresses)
-        self.resolve_dns_bootstrap_addresses()
-        return succeed(True)
+        if not self.initialized:
+            self.initialized = True
+            overlay.network.blacklist.extend(self.ip_addresses)
+            await self.resolve_dns_bootstrap_addresses()
```

Comment:
It seems to me that after the resolution of the DNS addresses, these should also be added to the `overlay.network.blacklist`. If that is correct, it is a bug in the original code. If you agree, could you fix that in this PR?

Reply:
Agreed. I'll fix it.

Reply:
On second thought, this appears to be happening already by calling `ensure_blacklisted` in `get_addresses`.